### PR TITLE
[pt1][quant] Unit Test for the Dynamic Quantized Linear operator

### PR DESCRIPTION
**Pull request by jianyuh**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23139 [pt1][quant] Unit Test for the Dynamic Quantized Linear operator**

Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

### CODE REVIEW

File: test/test_quantized.py [hunk 1]
```diff
@@ -315,6 +315,27 @@ def test_qlinear(self, use_bias, use_relu):
 
         torch.testing.assert_allclose(Y_fp32, Y_fp32_ref, rtol=0.0001, atol=1e-3)
 
+    def test_dynamic_qlinear(self):
```
Comment (zafartahirov) [inline test/test_quantized.py]: Hypothesis instead of randn?

Comment (jianyuh) [reply test/test_quantized.py]: Fixed.

File: test/test_quantized.py [hunk 2]
```diff
@@ -417,6 +417,93 @@ def test_qlinear(self, batch_size, input_channels, output_channels, use_bias, us
         self.assertEqual(Y_fp32, Y_fp32_ref,
                          message="torch.ops.quantized.fbgemm_linear_dynamic results are off")
 
+    @given(
+        batch_size=st.integers(1, 4),
+        input_channels=st.integers(16, 32),
+        output_channels=st.integers(4, 8),
+    )
+    def test_qlinear_legacy(self, batch_size, input_channels, output_channels):
```
Comment (raghuramank100) [inline test/test_quantized.py]: Is this a dup? I see the same code under test_qlinear in the same file in the other diff, please update

Comment (jianyuh) [reply test/test_quantized.py]: @raghuramank100 : This is for the unit test of the original `torch.fbgemm_linear_quantize_weight`, `torch.fbgemm_pack_quantized_matrix`, and `torch.fbgemm_linear_int8_weight` APIs for `torch.jit.quantized` workflow.

**Comment by jamesr66a** [comment]
@jianyuh do you want to land this?

**Comment by jianyuh** [comment]
@jamesr66a Either works. If we still want to keep track of the legacy QuantizedLinear ops (and if you approve this PR), we can land this and make sure the unit tests can cover more cases. Otherwise, we can just close this PR.

**Review by jamesr66a — APPROVED** [review]
Seems fine, more test coverage probably can't hurt

**Pull request closed without merging**